### PR TITLE
add login modal window in hall page, add 15 min timer

### PR DESCRIPTION
**Pull request by kds1660**
1. add timer directive/controller, when user add place to cart timer
start (15 minutes), when user empty cart - stop timer , add styles for timer
2. translate for login modal window, timer
3. for social add social button angular module with fonts
3. add modal login page with facebook for now login (send my app id for now, must change) and send login request when modal login window close, else if user dont login

### CODE REVIEW

File: src/css/hall.css [hunk 1]
```diff
@@ -249,3 +249,13 @@ li {
 .loggia-right li {
   transform: rotate(-80deg);
 }
+
+.modal-content {
+  background: #ffcdcd;
+  !important;
```
Comment (Varkh) [inline src/css/hall.css]: what is `!important` here?

File: src/js/Controllers/LoginCtrl.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        (function(){
+            if (document.getElementById('facebook-jssdk')) {return;}
+            var firstScriptElement = document.getElementsByTagName('script')[0];
+            var facebookJS = document.createElement('script');
+            facebookJS.id = 'facebook-jssdk';
+            facebookJS.src = '//connect.facebook.net/en_US/all.js';
+            firstScriptElement.parentNode.insertBefore(facebookJS, firstScriptElement);
+        }());
+    })
+    .controller('LoginCtrl', ['$scope', '$modalInstance','$rootScope', '$facebook', 'apiPost', 'userService',
```
Comment (Varkh) [inline src/js/Controllers/LoginCtrl.js]: move to different file: separate config and controller

File: src/js/Controllers/LoginCtrl.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            function refresh() {
+                $facebook.api("/me?fields=id,first_name,last_name,email").then(
+                    function(response) {
+                        userService.addUser(response);
+                        FB.getLoginStatus(function(response) {
+
+                            if (response.status === 'connected') {
+                                userService.setApiToken(response.authResponse.accessToken);
+                                userService.setNetwork('facebook');
+                                /*$rootScope.user.accessToken = response.authResponse.accessToken;
```
Comment (Varkh) [inline src/js/Controllers/LoginCtrl.js]: no comments in code. remove

File: src/js/Controllers/LoginCtrl.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                                $rootScope.user.network = 'facebook';*/
+                            } else {
+                                // the user isn't logged in to Facebook.
```
Comment (Varkh) [inline src/js/Controllers/LoginCtrl.js]: what should be here?

File: src/js/Controllers/hallCtrl.js [hunk 5]
```diff
@@ -26,6 +27,75 @@ angular.module('hall', [])
...
+            };
+
+            $modal.open({
+                templateUrl: 'views/shared/login.html',
+                windowClass: 'modal',
+                controller: 'LoginCtrl'
+            }).result.finally(function(){
+                //if (!$rootScope.user) $rootScope.user={}; //Todo change to object
+                apiPost('customers/login',{
+                    "first_name": userService.getCurrentUser().first_name, //Todo to service
```
Comment (Varkh) [inline src/js/Controllers/hallCtrl.js]: You can create ticket for all this TODOs, just don't left them unhandled

File: src/js/Controllers/hallCtrl.js [hunk 6]
```diff
@@ -26,6 +27,75 @@ angular.module('hall', [])
...
+                    timerInterval=0;
+                    $scope.timer=0;
+                }
+            };
+
+            $scope.$on('$destroy',function () {
+                $interval.cancel(timerInterval);
+            });
+
+            function getTimeRemaining(endtime){
```
Comment (Varkh) [inline src/js/Controllers/hallCtrl.js]: Move to some Util factory

**Review by Varkh — CHANGES_REQUESTED** [review]

File: src/css/hall.css (hunk 1 above)
Comment (kds1660) [reply src/css/hall.css]: owerwrite standart bootstrap modal white background, without bootstrap css priority highter

File: src/js/Controllers/LoginCtrl.js (hunk 4 above)
Comment (kds1660) [reply src/js/Controllers/LoginCtrl.js]: backend give me api token, thay dont implemented this yet...

File: src/js/Controllers/hallCtrl.js (hunk 5 above)
Comment (kds1660) [reply src/js/Controllers/hallCtrl.js]: I solved them, forget remove

File: src/js/Controllers/LoginCtrl.js (hunk 2 above)
Comment (kds1660) [reply src/js/Controllers/LoginCtrl.js]: in site config in module file,  I'm sticking to the logic of the site

File: src/js/Controllers/LoginCtrl.js (hunk 4 above)
Comment (kds1660) [reply src/js/Controllers/LoginCtrl.js]: i will remove this and call api get request, when modal window closed

File: src/js/Controllers/hallCtrl.js (hunk 6 above)
Comment (kds1660) [reply src/js/Controllers/hallCtrl.js]: Was going to do - timer service for hall and for shopping cart, now remake

**Review by Varkh — APPROVED** [review]

**Pull request merged**